homeassistant/components/roomba/config_flow.py

```diff
@@ -0,0 +1,136 @@
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+
+from . import CannotConnect, async_connect_or_timeout, async_disconnect_or_timeout
+from .const import (
+    CONF_BLID,
+    CONF_CERT,
+    CONF_CONTINUOUS,
+    CONF_DELAY,
+    CONF_NAME,
+    DEFAULT_CERT,
+    DEFAULT_CONTINUOUS,
+    DEFAULT_DELAY,
+    ROOMBA_SESSION,
+)
+from .const import DOMAIN  # pylint:disable=unused-import
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_BLID): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_CERT, default=DEFAULT_CERT): str,
+        vol.Optional(CONF_CONTINUOUS, default=DEFAULT_CONTINUOUS): bool,
+        vol.Optional(CONF_DELAY, default=DEFAULT_DELAY): int,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    roomba = Roomba(
+        address=data[CONF_HOST],
+        blid=data[CONF_BLID],
+        password=data[CONF_PASSWORD],
+        cert_name=data[CONF_CERT],
+        continuous=data[CONF_CONTINUOUS],
+        delay=data[CONF_DELAY],
+    )
+
+    info = await async_connect_or_timeout(hass, roomba)
+
+    return {
+        ROOMBA_SESSION: info[ROOMBA_SESSION],
+        CONF_NAME: info[CONF_NAME],
+        CONF_HOST: data[CONF_HOST],
+    }
+
+
+class RoombaConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Roomba configuration flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        import_info[CONF_BLID] = import_info.pop(CONF_USERNAME)
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors = {"base": "cannot_connect"}
+            except Exception:  # pylint: disable=broad-except
+                errors = {"base": "unknown"}
+
+            if "base" not in errors:
+                await async_disconnect_or_timeout(self.hass, info[ROOMBA_SESSION])
+                await self.async_set_unique_id(user_input[CONF_BLID])
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(title=info[CONF_NAME], data=user_input)
+
+        # If there was no user input, do not show the errors.
```

Comment:
```suggestion
        # If there was no user input, do not show the errors.
```

You can delete this block as you are already checking `user_input` is not None: